File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cs
public class Solution {
    public int FirstUniqChar(string s) {
        
        Dictionary<char,int> nonDuplicate = new Dictionary<char,int>();
        
        
        for(int i =0;i<s.Length;i++)
        {
            if(nonDuplicate.ContainsKey(s[i]) == false)
                nonDuplicate.Add(s[i],0);
            
           nonDuplicate[s[i]] ++;
        }
        
        
        for(int j = 0;j<s.Length;j++)
        {
            if(nonDuplicate[s[j]] == 1)
                return j;
        }
        return -1;
        
    }
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
          map<char,int> nonDuplicate ;
        
        
        for(int i =0;i<s.size();i++)
        {
            if( nonDuplicate.find(s[i]) == nonDuplicate.end() )
                nonDuplicate.insert(pair<char,int>(s[i],0) );
            
           nonDuplicate[s[i]] ++;
        }
        
        
        for(int j = 0;j<s.size();j++)
        {
            if(nonDuplicate[s[j]] == 1)
                return j;
        }
        return -1;
    }
};